### PR TITLE
refactor(i): Lint casts & move panics under `must` funcs

### LINKED ISSUE #3103: Fix StatusBadRequest -> StatusInternalServerError for internal errors in http handler
Suggestion from @nasdf:
https://github.com/sourcenetwork/defradb/pull/3099#discussion_r1787947741

Change: 
``` go
	db, ok := req.Context().Value(dbContextKey).(client.DB)
	if !ok {
		responseJSON(rw, http.StatusBadRequest, errorResponse{NewErrFailedToGetContext("db")})
		return
	}

```

To return error `500` so roughly this:
```go
	db, ok := req.Context().Value(dbContextKey).(client.DB)
	if !ok {
		responseJSON(rw, http.StatusInternalServerError, errorResponse{NewErrFailedToGetContext("db")})
		return
	}

```

### PR DESCRIPTION
**Pull request by shahzadlone**
## Relevant issue(s)
Resolves #3103 

## Description
- Fix the BadRequest in the acp handlers to now be panics
- Make the panics documented under `must` functions
- Add the linter to avoid increasing of undocumented panics
- Ignore linter in existing files outside cli and http (this should either be resolved gradually over time or in a single PR, regardless would prefer to not block this PR because of them)

### CODE REVIEW

File: tools/configs/golangci.yaml
Comment (AndrewSisley) [inline tools/configs/golangci.yaml:157]: question: Why would we want this?

**Review by AndrewSisley — CHANGES_REQUESTED** [review]
todo: This appears to be entirely untested, given the lack of any changes made to test files, and no note of any manual testing in the PR description.

I also have a question RE the linter excluded file list.

File: tools/configs/golangci.yaml
Comment (shahzadlone) [reply tools/configs/golangci.yaml:157]: Is kind of the last PR point "discussion" I wanted to hear opinions about. It's because there are many linter complaints in those files I didn't want to resolve in this PR. These should absolutely be all resolved so this list is empty.

Comment (AndrewSisley) [reply tools/configs/golangci.yaml:157]: Ah my bad, I didn't read the description carefully enough - I'm totally okay with this then, so long as this is documented in the comment, so it is very obvious to anyone reading that they should work towards reducing the list, not expanding it.

**Comment by AndrewSisley** [comment]
> HTTP handler functions changed to follow the same practice as all other existing handlers to panic instead of returning 400.

It was this I was mostly after - might not be worth the hassle of writing tests for though as I guess it is mostly just the http status code that changed, and that's not something the integration test framework can handle atm I think.  Have you manually tested this?

File: http/handler_acp.go
Comment (fredcarle) [inline http/handler_acp.go:44]: question: Why is this turned into a `must` instead of a `try` like what you did is some other handles?

File: http/middleware.go
Comment (fredcarle) [inline http/middleware.go:54]: thought: It feels wrong to have these in the `middleware` file. They should probably be moved to the `utils` file instead including the context variables above.

File: http/handler_tx.go
Comment (fredcarle) [inline http/handler_tx.go:73]: todo: This isn't a `BadRequest` but rather an `InternalServerError` (500). 

Out of scope of this PR but a lot of our returned statuses are wrong like this. I create #3139 to track this. If a panic within the handlers automatically returns a 500 status then it might be better to just do a `must` typecast and let the server panic for cases where a 500 is appropriate.

File: tools/configs/golangci.yaml
Comment (shahzadlone) [reply tools/configs/golangci.yaml:157]: Agreed will document that in the comment :)

- [ ] Document more clearly we need to reduce the list not expand it.

**Comment by shahzadlone** [comment]
> > HTTP handler functions changed to follow the same practice as all other existing handlers to panic instead of returning 400.
> 
> It was this I was mostly after - might not be worth the hassle of writing tests for though as I guess it is mostly just the http status code that changed, and that's not something the integration test framework can handle atm I think. Have you manually tested this?

Manually some yes, but not all. I am a bit unsure how to test certain situations.

File: http/handler_acp.go
Comment (shahzadlone) [reply http/handler_acp.go:44]: Only one that remained a `try` (not panic) is the P2P one because I think in that case it makes sense because P2P might be turned off and I see that specific error being returned by original author of the p2p code as useful for that situation as useful.

Rest all context casts are converted to a `must`.

File: http/middleware.go
Comment (shahzadlone) [reply http/middleware.go:54]: I agree with this thought.

(1) I was tempted to perhaps have a shared `utils` files some how for both `http` and `cli` as cli implements some of these methods too (however will have to make `http` and `cli` under a common parent folder perhaps that has the common `utils` file but we already have a `client` folder which I feel a bit uneasy moving both these into, perhaps another folder?). 

(2) Another approach might be to do what you suggested in a `utils` file similar to how it's done in `cli` package and call it a day.

Can follow consensus, does anyone else have any suggestions/prefs? @islamaliev @nasdf @AndrewSisley

File: http/handler_tx.go
Comment (shahzadlone) [reply http/handler_tx.go:73]: Actually #3139 seems similar / duplicate of what I wanted #3103 to be originally (if i understand correctly) . So I would say that is very much in the scope of this PR. I can change all the required `BadRequests` that need changing if you can guide me to which additional ones should be changed (@nasdf suggested only for the `context` gets+casts originally), I can take care of them in this PR.

File: http/middleware.go
Comment (AndrewSisley) [reply http/middleware.go:54]: I prefer (2) :)

Comment (shahzadlone) [reply http/middleware.go:54]: - [ ] Do (2)

File: tools/configs/golangci.yaml
Comment (shahzadlone) [reply tools/configs/golangci.yaml:159]: Resolved in commit: `PR(ADHERE): Adhere to code review - better documentation`

File: http/middleware.go
Comment (shahzadlone) [reply http/middleware.go]: Resolved in commit: `PR(ADHERE): Adhere to code review - move to utils`

File: http/handler_tx.go
Comment (nasdf) [reply http/handler_tx.go:73]: I agree, allowing a panic or changing to a 500 status would be better.

**Review by AndrewSisley — APPROVED** [review]
LGTM, but please make sure Keenan in particular is happy before merge :)

**Review by nasdf — APPROVED** [review]
LGTM assuming you fix Fred's remaining todo.

File: http/handler_tx.go
Comment (shahzadlone) [reply http/handler_tx.go:73]: > I agree, allowing a panic or changing to a 500 status would be better.

Just so I understand this, which one are you suggesting:
1) all the `ErrInvalidDataStoreTransaction` to panic?
2) all the `http.StatusBadRequest` to panic?
3) only this line the suggestion is on?

Comment (nasdf) [reply http/handler_tx.go:73]: Anywhere we do this type cast `txVal.(datastore.Txn)`.

Comment (shahzadlone) [reply http/handler_tx.go]: Done in commit: `PR(ADHERE): Adhere to code review - mustify datastore.Txn`

**Pull request merged**
